### ISSUE TITLE
plot_ensemble_model: OSError: syntax error in line 13 near '"'

### ISSUE THREAD

**Issue by kidotaka** [autogluon/autogluon#1065]
Hello
This is Windows related problem.
I know it's not supported yet.

TabularPredictor#plot_ensemble_model() result in OSError from pygraphviz on Windows.

A line number in the message vary by dot file and it's not an actual problem line.

Each node have "path" attribute and include backslash separator on Windows.

It seems that backslash is treated as an escape.

After replacing backslash to slash, I can plot it.

**Comment by GaoyangLiu** [comment]
Same error on Windows 10.
AutoGluon:    0.4.2
pygraphviz:    1.9
```
---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
~\AppData\Local\Temp\ipykernel_25620\377960091.py in <cell line: 1>()
----> 1 predictor.plot_ensemble_model()

C:\ProgramData\Anaconda3\envs\ag10\lib\site-packages\autogluon\tabular\predictor\predictor.py in plot_ensemble_model(self, prune_unused_nodes)
   2665         model_image_fname = os.path.join(self.path, 'ensemble_model.png')
   2666 
-> 2667         A.draw(model_image_fname, format='png', prog='dot')
   2668 
   2669         return model_image_fname

C:\ProgramData\Anaconda3\envs\ag10\lib\site-packages\pygraphviz\agraph.py in draw(self, path, format, prog, args)
   1609             args = " ".join([args, "-T" + format])
   1610 
-> 1611         data = self._run_prog(prog, args)
   1612 
   1613         if path is not None:

C:\ProgramData\Anaconda3\envs\ag10\lib\site-packages\pygraphviz\agraph.py in _run_prog(self, prog, args)
   1400 
   1401         if not data:
-> 1402             raise OSError(b"".join(errors).decode(self.encoding))
   1403 
   1404         if len(errors) > 0:

OSError: Error: <stdin>: syntax error in line 25 near '"'
```

With exactly the same code, it works fine in linux.

**Comment by HenryPeking** [comment]
Same Error on Windows 11.
AutoGluon 0.7
pygraphviz: 1.9

in AGraph._run_prog(self, prog, args)
   1399 p.wait()
   1401 if not data:
-> 1402     raise OSError(b"".join(errors).decode(self.encoding))
   1404 if len(errors) > 0:
   1405     warnings.warn(b"".join(errors).decode(self.encoding), RuntimeWarning)

**Comment by kevintsq** [comment]
@kidotaka Could you please explain where to replace backslash to slash? Thanks.

**Comment by kevintsq** [comment]
> > @kidotaka Could you please explain where to replace backslash to slash? Thanks.
> 
> @kevintsq https://github.com/autogluon/autogluon/blob/master/tabular/src/autogluon/tabular/predictor/predictor.py#L2970 In the loop of nodes, replace backslash of a 'path' attribute to slash if a node has a 'path' attribute.

Thanks.
```python
            node.attr['path'] = node.attr['path'].replace('\\', '/')
```

**Comment by Innixma** [comment]
This should be resolved by #3259 in the upcoming v0.8 release.
